pkg/signature/kms/cliplugin/client.go

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cliplugin implements the plugin functionality.
+package cliplugin
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"strings"
+
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/sigstore/sigstore/pkg/signature/kms"
+	"github.com/sigstore/sigstore/pkg/signature/kms/cliplugin/common"
+)
+
+const (
+	// PluginBinaryPrefix is the prefix for all plugin binaries. e.g., sigstore-kms-my-hsm.
+	PluginBinaryPrefix = "sigstore-kms-"
+)
+
+// init registers the plugin system as a provider. It does not search for plugin programs.
+// users must import this package, e.g.,  `import _ "github.com/sigstore/sigstore/pkg/signature/kms/cliplugin"`
+func init() {
+	kms.AddProvider(kms.CLIPluginProviderKey, func(ctx context.Context, keyResourceID string, hashFunc crypto.Hash, opts ...signature.RPCOption) (kms.SignerVerifier, error) {
+		return LoadSignerVerifier(ctx, keyResourceID, hashFunc, opts...)
+	})
+}
+
+// LoadSignerVerifier creates a PluginClient with these InitOptions.
+func LoadSignerVerifier(ctx context.Context, inputKeyresourceID string, hashFunc crypto.Hash, opts ...signature.RPCOption) (kms.SignerVerifier, error) {
```

Comment:
Because it has a vault token? 
* https://github.com/ramonpetgrave64/sigstore/blob/185deaa2f5438a6e8ba828f625fee29724e3895c/pkg/signature/options/rpcauth.go#L28

I agree about not putting secrets in CLI, But I think we should keep `RPCOption`, because it has the `Context` and `KeyVersion`, and probably drop `options.RPCAuth`.
* https://github.com/ramonpetgrave64/sigstore/blob/185deaa2f5438a6e8ba828f625fee29724e3895c/pkg/signature/options.go#L31

And the hashivault package, is already written to support reading creds from env, same as the other KMS packages.